### DISCUSSION TITLE
在macOS/Safari环境上无法过cloudflare的人机验证

### DISCUSSION THREAD

**Discussion by laye0619** [immortalwrt/homeproxy#155] [Q&A] [answered]
参考：https://rushb.pro/article/openclash-cloudflare_challenges.html

文章中指出的问题就是我现在遇到的问题，但是文章是基于clash给出的解决方案（方法三），我在homeproxy中设置结果没有作用。但我相信原理是相通的，即：
- 加入苹果的几个域名到中国ipset，使得当勾选了“绕过中国流量”后，可以让这几个域名不走homeproxy/sing-box内核，直接通过防火墙走direct。

补充一下，我曾经看了一下openwrt的防火墙。将上述链接中的几个苹果域名加入到了ipset（网络->DHCP/DNS->IP集）：@homeproxy_mainland_addr_v4 中，但是还是没有啥作用，人机验证还是没反应。
<img width="1014" alt="image" src="https://github.com/user-attachments/assets/1c751274-085b-4fe7-b4a4-3118963a0392">

诚心求教哈～

**Comment by muink** [comment]
我觉得只是dnsmasq的自动添加ipset没在工作而已
`/var/etc/dnsmasq.conf` 里已经有 `nftset=/mask.icloud.com/mask-h2.icloud.com/mask.apple-dns.net/4#ip#fw4#homeproxy_wan_direct_addr_v4`
但是在 dig 或 nslookup 后 并不会自动添加

**Reply by laye0619** [reply]
请教这应该怎么修改才能达到目的呢？

**Reply by muink** [reply]
~查到原因了, dnsmasq 的 init.d 脚本有问题~
更正: config 中 table_family 设置问题, fw4 的 familys 是 inet, 只能设置 inet 不能设置 ip 或 ip6

**Reply by laye0619** [reply]
烦请告知应该怎么修改？我确实不是技术背景……，十分感谢！

**Comment by muink** [comment, marked as answer]
@laye0619 
`/etc/config/dhcp` 里 ipset 设置成这样, set 名不要加 @ 号
```
config ipset
	option table_family 'ip'
	list name 'homeproxy_wan_direct_addr_v4'
	list domain 'mask.icloud.com'
	list domain 'mask-h2.icloud.com'
	list domain 'mask.apple-dns.net'
```
init.d 脚本不用改了, 你把前面这段的配置的 table_family 改成 inet 就能工作了
```
	option table_family 'inet'
```

**Reply by laye0619** [reply]
已经改成inet了（openwrt也重启了），人机验证还是过不了，很奇怪
<img width="434" alt="image" src="https://github.com/user-attachments/assets/b98ecf1c-686b-4988-aa49-74488bfc09f1">

**Reply by muink** [reply]
自查下 `nft list set inet fw4 homeproxy_mainland_addr_v4`

**Reply by laye0619** [reply]
$ nft list set inet fw4 homeproxy_mainland_addr_v4 | grep "icloud"   没有结果……

**Reply by muink** [reply]
这里只能找到 ip 找不到 domain
你看看 elements 里有没有你刚才解析出来的 ip 就好了

**Reply by laye0619** [reply]
我已经焦头烂额，暂时使用tun代理udp了，这样就可以过人机验证了。不过还是十分感谢你哈！
